rrs/tools/upgrade_history_internal.py

```diff
@@ -0,0 +1,617 @@
+    urldir = str(layer.get_fetch_dir())
+    repodir = os.path.join(fetchdir, urldir)
+    layerdir = os.path.join(repodir, str(layerbranch.vcs_subdir))
+
+    if layerbranch.vcs_subdir:
+        layersubdir_start = layerbranch.vcs_subdir
+        if not layersubdir_start.endswith('/'):
+            layersubdir_start += '/'
+    else:
+        layersubdir_start = ''
+
+    repo = git.Repo(repodir)
+    if repo.bare:
+        logger.error('Repository %s is bare, not supported' % repodir)
+        sys.exit(1)
+
+    commitdate = checkout_layer_deps(layerbranch, commit, fetchdir, logger)
+
+    if options.initial:
+        fns = None
+        deleted = []
+        moved = []
+    else:
+        if options.filter_files:
+            filepath_start = options.filter_files
+        else:
+            filepath_start = layersubdir_start
+        fns, deleted, moved = _get_recipes_filenames(commit, repo, repodir, filepath_start, logger)
+        if not (fns or deleted or moved):
+            return
+
+    # setup bitbake
+    bitbakepath = os.path.join(fetchdir, 'bitbake')
+    if options.bitbake_rev:
+        bitbake_rev = options.bitbake_rev
+        if not re.match('^[0-9a-f]{40}$', bitbake_rev):
+            # Branch name, need to check out detached
+            bitbake_rev = 'origin/%s' % bitbake_rev
+    else:
+        bitbake_rev = utils.runcmd(['git', 'rev-list', '-1', '--before=%s' % commitdate, 'origin/master'], bitbakepath, logger=logger).strip()
+    utils.checkout_repo(bitbakepath, bitbake_rev, logger)
+    sys.path.insert(0, os.path.join(bitbakepath, 'lib'))
+
+    (tinfoil, d, recipes, tempdir) = load_recipes(layerbranch, bitbakepath,
+                        fetchdir, settings, logger, recipe_files=fns,
+                        nocheckout=True)
+    try:
+
+        if options.initial:
+            title = options.initial
+            info = 'No maintainer;;' + utils.runcmd(['git', 'log', '--format=%ad;%cd', '--date=rfc', '-n', '1', commit], destdir=repodir, logger=logger)
+            recordcommit = ''
+        else:
+            title = utils.runcmd(['git', 'log', '--format=%s', '-n', '1', commit],
+                                            repodir, logger=logger)
+            info = utils.runcmd(['git', 'log', '--format=%an;%ae;%ad;%cd', '--date=rfc', '-n', '1', commit], destdir=repodir, logger=logger)
+            recordcommit = commit
+
+        fn_data = {}
+        for recipe_data in recipes:
+            fn = os.path.relpath(recipe_data.getVar('FILE', True), repodir)
+            fn_data[fn] = recipe_data
+
+        seen_pns = []
+        try:
+            with transaction.atomic():
+                # Handle recipes where PN has changed
+                for a, b in moved:
+                    logger.debug('Move %s -> %s' % (a,b))
+                    rus = RecipeUpgrade.objects.filter(recipesymbol__layerbranch=layerbranch, filepath=a).order_by('-commit_date', '-id')
+                    recipe_data = fn_data.get(b, None)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are using the `get()` method to retrieve the value for a key from a dictionary. If the key is not present, then `get()` returns the value that is passed in its optional second parameter. If if the key is not present, then `None` is the default return value. Therefore, there is no need to explicitly specify `None` in the second parameter.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)